test/functional/wallet_orphanedreward.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test orphaned block rewards in the wallet."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, assert_raises_rpc_error
+
+class OrphanedBlockRewardTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.supports_cli = False
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        # Generate some blocks and obtain some coins on node 0.  We send
+        # some balance to node 1, which will hold it as a single coin.
+        self.nodes[0].generate(150)
+        self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 10)
+        self.nodes[0].generate(50)
+
+        # Get a block reward with node 1 and remember the block so we can orphan
+        # it later.
+        self.sync_blocks()
+        blk = self.nodes[1].generate(1)[0]
+        self.sync_blocks()
+
+        # Let the block reward mature and send coins including both
+        # the existing balance and the block reward.
+        self.nodes[0].generate(150)
+        assert_equal(self.nodes[1].getbalance(), 10 + 25)
+        self.nodes[1].sendtoaddress(self.nodes[0].getnewaddress(), 30)
+
+        # Orphan the block reward and make sure that the original coins
+        # from the wallet can still be spent.
+        self.nodes[0].invalidateblock(blk)
+        self.nodes[0].generate(200)
+        self.sync_blocks()
+        self.nodes[1].sendtoaddress(self.nodes[0].getnewaddress(), 9)
```

Comment:
Could comment this out, and add a comment that it fails?

Reply:
Instead of this, I've now added in the `abandontransaction` call that makes the test succeed (but according to #14148 should not be needed) as well as a comment explaining this.  I think that makes the situation even more clear.